controller/attribute/util.py

```diff
@@ -73,24 +83,29 @@ def run_attribute_calculation_exec_env(
         attribute_item.data_type,
     ]
 
-    container = client.containers.run(
+    # initial log preparation
+    __update_progress(project_id=project_id, attribute_item=attribute_item, progress=0.0)
+
+    container_name = str(uuid.uuid4())
+    container = client.containers.create(
         image=image,
         command=command,
-        remove=True,
+        auto_remove=True,
         detach=True,
         network=exec_env_network,
     )
-
-    logs = [
+    __containers_running[container_name] = True
+    daemon.run(__read_container_logs_thread, project_id, container_name, attribute_id, container)
+    container.start()
+    attribute_item.logs = [
         line.decode("utf-8").strip("\n")
         for line in container.logs(
             stream=True, stdout=True, stderr=True, timestamps=True
         )
+        if "progress" not in line.decode("utf-8")
     ]
-
-    attribute.update(
-        project_id=project_id, attribute_id=attribute_id, logs=logs, with_commit=True
-    )
+    del __containers_running[container_name]
+    __update_progress(project_id, attribute_item, 1.0) 
```

Comment:
the attribute calculation isn't finished with the exec environment. Thats why in the lf logic we set it to 0.05 before starting the container, 0.9 after the container is finished and every progress from the container (0-1) multiplied with 0.8. Why was decided to skip this part here? For me now the progress bar looks finished (with 100%) but still takes quite some time to get the calculated valued to the database.

Reply:
I did not skip the part, I just built a threaded solution which sends the progress of the execution env from 0.0 to 1.0. But ofc I can change it to the behavior you described